### DISCUSSION TITLE
Rotated Hovmöller diagram

### DISCUSSION THREAD

**Discussion by SimonPeatman** [SciTools/iris#4940] [Q&A] [answered]
Suppose I have a Cube of the form:
```
pcp / (mm hr-1)                     (time: 96; latitude: 128; longitude: 93)
    Dimension coordinates:
        time                             x             -               -
        latitude                         -             x               -
        longitude                        -             -               x
    Auxiliary coordinates:
        x                                -             x               x
        y                                -             x               x
```
The multi-dimensional ```x``` and ```y``` AuxCoords are the coordinates of a rectangular grid, but it is rotated relative to lines of latitude and longitude.  Ultimately, I want to select a particular range of ```x``` and ```y``` values, then average over ```y``` to create a Hovmöller diagram in ```(time, x)```.

The only way I know of doing this is to create arrays of the ```latitude``` and ```longitude``` values at the points I want on the rotated grid, interpolate to these points using ```iris.analysis.trajectory.interpolate()```, reshape the result, then average over the resulting (single-dimensional) ```y``` direction.  However, the interpolation is a slow step.

Is there any other way of achieving this?  Mathematically, it must be possible to do some clever averaging over part of the original Cube without having to interpolate to the rotated grid first, but I don't know if an implementation of this exists.

**Comment by trexfeathers** [comment, marked as answer]
Hi @SimonPeatman 🙂

First of all: we've just released [a 50x speed-up to `iris.analysis.trajectory.interpolate()`](https://scitools-iris.readthedocs.io/en/v3.3.0/whatsnew/3.3.html#performance), if that's any help to you.

Could you also create new 1-dimensional coordinates for your rotated system using [`cartopy.crs.CRS`](https://scitools.org.uk/cartopy/docs/latest/reference/generated/cartopy.crs.CRS.html#cartopy.crs.CRS)`.transform_points()`, then use the standard 1-dimensional logic?

**Reply by pp-mo** [reply]
No further reply, so I'll mark this as answered + move to "Archive"
Please re-raise if needed !
